src/vm/method.cpp

```diff
@@ -4597,23 +4708,56 @@ static_assert_no_msg((TEP_CENTRAL_JUMP_SIZE & 1) == 0);
     // Always do consistency check in debug
     if (fSpeculative INDEBUG(|| TRUE))
     {
+#ifdef _TARGET_ARM_
+        if (!IsCompactEntryPointAtAddress(addr))
+#else // _TARGET_ARM_
         if ((addr & 3) != 1 ||
             *PTR_BYTE(addr) != X86_INSTR_MOV_AL ||
             *PTR_BYTE(addr+2) != X86_INSTR_JMP_REL8)
+#endif // _TARGET_ARM_
         {
             if (fSpeculative) return NULL;
             _ASSERTE(!"Unexpected code in temporary entrypoint");
         }
     }
 
+#ifdef _TARGET_ARM_
+
+    // On ARM compact entry points are thumb
+    _ASSERTE ((addr & THUMB_CODE) != 0);
+    addr = addr - THUMB_CODE;
+
+    // Get offset for CentralJumpCode from current compact entry point
+    uint16_t *pBranchInstr = ((uint16_t *)addr) + 1;
```

Comment:
This needs to be DACized properly so that it does not break debugger: cast to `PTR_XXX` and dereference that, similar to how x86 is done.

More info about DAC: https://github.com/dotnet/coreclr/blob/master/Documentation/botr/dac-notes.md